src/jv.c

```diff
@@ -995,7 +995,12 @@ jv jv_array_set(jv j, int idx, jv val) {
   // copy/free of val,j coalesced
   jv* slot = jvp_array_write(&j, idx);
   jv_free(*slot);
-  *slot = val;
+  if(jv_is_borrowed(val)){
+    *slot = jv_unborrow(val);
```

Comment:
Aha, see, the elements of arrays aren't borrowed, so no need to unborrow when reading them.

Reply:
But we are writing val to *slot. We need to increment the refcount and unborrow there

Reply:
Yes, but we don't need to unborrow when _reading_ the slot elsewhere _because_ it's always not-borrowed once written into the slot.

Reply:
I'm not saying _this_ code here should change.  I'm saying the readers don't need to unborrow what they find in arrays/objects/invalids -- they're already not borrowed.